backend/app/parsers.py

```diff
@@ -1,39 +1,77 @@
 from flask_restful import reqparse
-from sqlalchemy import exists
 
-from app.models import Category, Subcategory
+from app.models import Category, Subcategory, Item, User
 from init_app import db
 
 
-def check_category(value):
-    if not db.session.query(exists().where(Category.name == value)).scalar():
-        raise ValueError("Value is not Category instance")
+def object_exist(model, value, name):
+    user_attribute = getattr(model, name)
+    return db.session.query(model.query.filter(user_attribute == value).exists()).scalar()
+
+
+def check_object(model, value, name='name'):
+    if not object_exist(model, value, name):
+        raise ValueError(f"{value} is not class instance")
     return value
 
 
-def check_subcategory(value):
-    if not db.session.query(exists().where(Subcategory.name == value)).scalar():
-        raise ValueError("Value is not Subcategory instance")
+def check_unique_object(model, value, name='name'):
+    if object_exist(model, value, name):
+        raise ValueError(f"{value} already exist")
     return value
 
 
+def check_category(value):
```

Comment:
Try to create this function with `partial` from `functools` package :) It should be in form as below:
`check_category = partial(...)`

Try to do this :)